src/backend/executor/bridge.ts

```diff
@@ -107,21 +108,36 @@ export class ExecutorBridge implements Disposable {
         return undefined;
     }
 
-    public getAnalyzeCmdLine(...files: Uri[]): string | undefined {
+    public getAnalyzeCmdArgs(...files: Uri[]): string[] | undefined {
         if (!workspace.workspaceFolders?.length) {
             return undefined;
         }
 
         // TODO: Refactor for less code repetition across functions
         const workspaceFolder = workspace.workspaceFolders[0].uri.fsPath;
 
-        const ccPath = getConfigAndReplaceVariables('codechecker.executor', 'executablePath')
-            ?? 'CodeChecker';
         const ccFolder = getConfigAndReplaceVariables('codechecker.backend', 'outputFolder')
             ?? path.join(workspaceFolder, '.codechecker');
-        const ccArguments = getConfigAndReplaceVariables('codechecker.executor', 'arguments') ?? '';
-        const ccThreads = workspace.getConfiguration('codechecker.executor').get<string>('threadCount');
 
+        const ccArgumentsSetting = workspace.getConfiguration('codechecker.executor').get<string>('arguments');
+
+        // TODO: Merge this collection with replaceVariables
+        const env: { [key: string]: string } = {
+            workspaceFolder,
+            workspaceRoot: workspaceFolder,
+            cwd: process.cwd()
+        };
+        for (const [key, val] of Object.entries(process.env)) {
+            if (val !== undefined) {
+                env[`env.${key}`] = val;
```

Comment:
Shouldn't it be `env[key] = val` instead?

Reply:
The default VS Code settings handle environment variables a bit differently [(docs)](https://code.visualstudio.com/docs/editor/variables-reference), so the env variables are accessible via ``${env:<env>}`` in eg. other settings as well.

VSCode doesn't provide an auto-substitute function, that's the only reason it is done manually here.

(There's also a typo here, `env.` instead of `env:`, fixed.)

Reply:
Thanks. It looks correct.